Makefile

```diff
@@ -289,6 +290,9 @@ distclean: clean assetclean
 
 setup:
 	$(MAKE) -C tools
+	test -d $(VENV) || python3 -m venv $(VENV)
+	$(PYTHON) -m pip install -U pip
+	$(PYTHON) -m pip install -r requirements.txt
```

Comment:
Thoughts on maybe doing something like this instead
```make
$(VENV): requirements.txt
	python3 -m venv $(VENV)
	$(PYTHON) -m pip install -U pip
	$(PYTHON) -m pip install -U -r requirements.txt
	touch $(VENV)

setup: $(VENV)
	$(MAKE) -C tools
	$(PYTHON) tools/decompress_baserom.py $(VERSION)
	$(PYTHON) extract_baserom.py
	$(PYTHON) extract_assets.py -j$(N_THREADS)
```
this way it keeps all the venv logic together, and would only update if requirements.txt gets updated. This also gives a way to just build/update the venv without needing a full setup again.

Reply:
I thought directories behave badly if you use them as makefile targets, but I do like the idea of splitting out the logic into a separate target, and depending on requirements.txt. Maybe we can have the target be `.venv/pyvenv.cfg` instead, or our own sentinel file?

Reply:
Reading up on this a bit more, I think `setup: | $(VENV)` would work, I'll try it out

Reply:
Personally I'm fine with a sentinel file (would prefer making our own though). Though in that case we may want to add a phony `venv` target that depends on it, just to make setting up/updating the venv without a full setup a little easier (though maybe not necessary).

Reply:
Okay, I created 2 separate targets, one for creating the venv (which we only want to do once ever) and one for installing python deps if `requirements.txt` changes. I'm also not sure what to call these things so please suggest better names